From e3e9a3a6005b1e70d355a6dc392e1a039188c0cf Mon Sep 17 00:00:00 2001
From: Nico Mexis <nico.mexis@kabelmail.de>
Date: Sun, 3 Nov 2024 20:59:53 +0100
Subject: [PATCH] Fix BOOST detection on MSYS2/UCRT

This is based on code provided previously by @JohnAAbbott - maybe it was removed by accident?
---
 configuration/boost-find-lib.sh | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/configuration/boost-find-lib.sh b/configuration/boost-find-lib.sh
index 032b00c..3f20d35 100755
--- a/configuration/boost-find-lib.sh
+++ b/configuration/boost-find-lib.sh
@@ -11,7 +11,8 @@ SCRIPT_NAME=[[$(basename "$0")]]
 # 2022-09-14: previously needed "system thread filesystem"
 ### 2022-11-12: my SmallBlue computer seems to need "system" to build the GUI
 # BOOST 1.67 and earlier need both filesystem and system
-SUBLIBS="filesystem  system"   # ORDER IS IMPORTANT!  system needed for older BOOST (before 1.74)
+# 2024-10-02: Nico Mexis reports that CygWin uses boost 1.66
+SUBLIBS="filesystem"   # ORDER IS IMPORTANT!  system needed for older BOOST (before 1.74) -- see below (we include system if it is found next to libboost_filesystem)
 
 # taken from StackExchange 256434
 is_absolute()
@@ -145,6 +146,15 @@ then
   exit 1
 fi
 
+# CHEAP HACK: hopefully it can soon be eliminated (when older systems
+#             update to newer versions of BOOST)
+# Older versions of BOOST must also link to libboost_system
+/bin/ls "$BOOST_LIB_DIR"/libboost_system.*  >/dev/null  2>&1
+if [ $? = 0 ]
+then
+    SUBLIBS="$SUBLIBS  system"
+fi
+
 
 # Create symlinks to the libraries found, and put the names in BOOST_LDLIBS
 BOOST_LDLIBS=""
